test/xspec-space_stylesheet.xspec

```diff
@@ -0,0 +1,384 @@
+					resolve-QName($qname, /x:description))],</x:label>
+				<x:expect label="whitespace-only text nodes are kept."
+					test="deep-equal($global-param-pre-element, $pre-element-wsot)" />
+			</x:scenario>
+		</x:scenario>
+	</x:scenario>
+
+	<x:scenario
+		label="In assertion, whitespace-only text nodes in user-content are removed by default.">
+		<x:scenario>
+			<x:label>So, in //x:expect/element()[namespace-uri() ne
+				'http://www.jenitennison.com/xslt/xspec'][not(ancestor-or-self::element()/@xml:space)],</x:label>
+			<x:call function="exactly-one">
+				<x:param select="$span-element-empty" />
+			</x:call>
+			<x:expect label="whitespace-only text nodes are removed.">
+				<span>&#x09;&#x0A;&#x0D;&#x20;</span>
+			</x:expect>
+		</x:scenario>
+
+		<x:scenario label="But...">
+			<x:scenario>
+				<x:label>@xml:space overrides the default. So, in
+					//x:expect/element()[namespace-uri() ne
+					'http://www.jenitennison.com/xslt/xspec'][ancestor-or-self::element()[@xml:space][1]/@xml:space
+					= "preserve"],</x:label>
+				<x:call function="exactly-one">
+					<x:param select="$span-element-wsot" />
+				</x:call>
+				<x:expect label="whitespace-only text nodes are kept." xml:space="preserve"><span>&#x09;&#x0A;&#x0D;&#x20;</span></x:expect>
+			</x:scenario>
+
+			<x:scenario>
+				<x:label>node-selection @href is always intact. So, in
+					//x:expect/@href[not(ancestor::element()/@xml:space)]/doc(.)[not(descendant::element()/@xml:space)],</x:label>
+				<x:call function="exactly-one">
+					<x:param select="$span-element-wsot" />
+				</x:call>
+				<x:expect href="xspec-space.xml" label="whitespace-only text nodes are kept."
+					select="node()" />
+			</x:scenario>
+
+			<x:scenario>
+				<x:label>Elements specified by @preserve-space are intact. So, in
+					//x:expect/element()[not(ancestor-or-self::element()/@xml:space)][node-name(.) =
+					(for $qname in tokenize(/x:description/@preserve-space, '\s+') return
+					resolve-QName($qname, /x:description))],</x:label>
+				<x:call function="exactly-one">
+					<x:param select="$pre-element-wsot" />
+				</x:call>
+				<x:expect label="whitespace-only text nodes are kept.">
+					<pre>&#x09;&#x0A;&#x0D;&#x20;</pre>
+				</x:expect>
+			</x:scenario>
+		</x:scenario>
+	</x:scenario>
+
+	<x:scenario label="In user-content,">
+		<x:scenario>
+			<x:label>@xml:space="preserve" takes effect on descendant whitespace-only text node.
+				So...</x:label>
+			<x:call function="exactly-one">
+				<x:param>
+					<p xml:space="preserve">&#x09;&#x0A;&#x0D;&#x20;<span>&#x09;&#x0A;&#x0D;&#x20;</span></p>
+				</x:param>
+			</x:call>
+			<x:expect label="Child whitespace-only text nodes are kept."
+				test="deep-equal(p/(node() except span), $wsot)" />
+			<x:expect label="Grandchild whitespace-only text nodes are kept."
+				test="deep-equal(p/span, $span-element-wsot)" />
+		</x:scenario>
+
+		<x:scenario>
+			<x:label>Elements specified by @preserve-space takes effect only on child
```

Comment:
The grammar in the label is a little off. Alternate suggestions you might consider: `Elements specified by @preserve-space see effect only on...` or `For elements specified by @preserve-space, effect is only on...`.

Reply:
Thanks for the suggestions. I took the latter in ea6ee7734f5958c2c7b8bf18514779ccb64e2ede.